grid_map_filters/src/grid_map_filters/NormalVectorsFilter.cpp

```diff
@@ -170,52 +202,60 @@ void NormalVectorsFilter<T>::computeWithAreaSerial(GridMap& map, const std::stri
     }
   }
 
-  const double end = ros::Time::now().toSec();
-  ROS_DEBUG_THROTTLE(2.0, "NORMAL COMPUTATION TIME = %f", (end - start));
+  const double end = clock.now().seconds();
+  RCLCPP_DEBUG_THROTTLE(
+    logging_interface_->get_logger(), clock, 2.0, "NORMAL COMPUTATION TIME = %f",
+    (end - start));
 }
 
-template <typename T>
-void NormalVectorsFilter<T>::computeWithAreaParallel(GridMap& map, const std::string& inputLayer, const std::string& outputLayersPrefix) {
-  const double start = ros::Time::now().toSec();
+void NormalVectorsFilter::computeWithAreaParallel(
+  GridMap & map, const std::string & inputLayer,
+  const std::string & outputLayersPrefix)
+{
+  rclcpp::Clock clock;
+  const double start = clock.now().seconds();
   grid_map::Size gridMapSize = map.getSize();
 
-  // Set number of thread to use for parallel programming.
-  std::unique_ptr<tbb::task_scheduler_init> TBBInitPtr;
-  if (threadCount_ != -1) {
-    TBBInitPtr.reset(new tbb::task_scheduler_init(threadCount_));
-  }
-
   // Parallelized iteration through the map.
```

Comment:
Can you add a comment here that the number of threads is automatically determined?